pkg/handler/handler.go

```diff
@@ -175,6 +176,77 @@ func (h *Handler) GetApiV1OrganizationsOrganizationIDRegionsRegionIDImages(w htt
 	util.WriteJSONResponse(w, r, http.StatusOK, result)
 }
 
+func (h *Handler) PostApiV1OrganizationsOrganizationIDRegionsRegionIDImages(w http.ResponseWriter, r *http.Request, organizationID openapi.OrganizationIDParameter, regionID openapi.RegionIDParameter) {
+	if err := rbac.AllowOrganizationScope(r.Context(), "region:images", identityapi.Create, organizationID); err != nil {
+		errors.HandleError(w, r, err)
+		return
+	}
+
+	var request openapi.ImageCreateRequest
+	if err := util.ReadJSONBody(r, request); err != nil {
+		errors.HandleError(w, r, err)
+		return
+	}
+
+	result, err := region.NewClient(h.client, h.namespace).CreateImage(r.Context(), organizationID, regionID, &request)
+	if err != nil {
+		errors.HandleError(w, r, err)
+		return
+	}
+
+	util.WriteJSONResponse(w, r, http.StatusOK, result)
+}
+
+func (h *Handler) DeleteApiV1OrganizationsOrganizationIDRegionsRegionIDImagesImageID(w http.ResponseWriter, r *http.Request, organizationID openapi.OrganizationIDParameter, regionID openapi.RegionIDParameter, imageID openapi.ImageIDParameter) {
+	if err := rbac.AllowOrganizationScope(r.Context(), "region:images", identityapi.Delete, organizationID); err != nil {
+		errors.HandleError(w, r, err)
+		return
+	}
+
+	if err := region.NewClient(h.client, h.namespace).DeleteImage(r.Context(), organizationID, regionID, imageID); err != nil {
+		errors.HandleError(w, r, err)
+		return
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+}
+
+func (h *Handler) PostApiV1OrganizationsOrganizationIDRegionsRegionIDImagesImageID(w http.ResponseWriter, r *http.Request, organizationID openapi.OrganizationIDParameter, regionID openapi.RegionIDParameter, imageID openapi.ImageIDParameter) {
+	if err := rbac.AllowOrganizationScope(r.Context(), "region:images", identityapi.Create, organizationID); err != nil {
+		errors.HandleError(w, r, err)
+		return
+	}
+
+	// FIXME: Uncomment this line and decide on a maximum size for the upload content.
+	// r.GetBody = http.MaxBytesReader(w, r.Body, TO_BE_DECIDED)
+
+	// Even though we are passing 0 into this ParseMultipartForm function call, the underlying implementation adds an extra 10MiB buffer by default.
+	// Only the first 10MiB of the file will be read into memory, and any additional data beyond that will be streamed to disk.
```

Comment:
This is OK for now. I wonder if we can open the request to OpenStack and give it a truncated reader from here, or it's too much trouble.